### PR TITLE
Introduce new poller functionalities and new inventory format

### PR DESCRIPTION
**Pull request by claudiolor**
This PR Introduces a bunch of improvements to the poller:
- Add dynamic inventories support
- Add source plugin system
- Add Netbox source
- New unique inventory file format specified with -I argument
- Launch multiple instances of the poller workers

### CODE REVIEW

File: suzieq/poller/controller/controller.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+    @property
+    def inventory_timeout(self) -> int:
+        """Returns the maximum amount of time to wait for an inventory source
+        retrieving its device list.
+
+        Returns:
+            int: inventory timeout in secodns
+        """
+        return self._inventory_timeout
```
Comment (ddutt) [inline suzieq/poller/controller/controller.py:142]: Does this need to be inventory source specific? For now, I think not. But don't be surprised if you get asked that

**Review by ddutt — CHANGES_REQUESTED** [review]
I'm going to merge this anyways. But if you could address the user config comments, that'd be great.

File: suzieq/poller/controller/controller.py (hunk 1 above)
Comment (LucaNicosia) [reply suzieq/poller/controller/controller.py:142]: There is a timeout for each source which collects the inventory from the source and a single timeout for the controller which collects new inventories from sources and create the global inventory

File: suzieq/config/etc/suzieq-cfg.yml [hunk 2]
```diff
@@ -30,7 +30,17 @@ poller:
...
+    path: "suzieq/.poller/intentory/inventory.yaml"
+  manager:
+    type: static_manager
+    # workers_count: 1
+    # inventory_path: suzieq/.poller/intentory/static_inventory
+    # inventory_file_name: static_inv
+    # start_workers: False
+  chunker:
+    type: static_chunker
+    split_pol: namespace
```
[the remark this answers is not in the document]
Comment (LucaNicosia) [reply suzieq/config/etc/suzieq-cfg.yml:43]: This is the chunking policy. For the alpha I'll drop the option and doesn't allow the user to add the policy

File: suzieq/poller/controller/base_controller_plugin.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        ptype = plugin_conf.get("type")
+        if not ptype:
+            raise SqPollerConfError('No default type provided')
+
+        controller_class = cls.get_plugins(plugin_name=ptype)
+
+        if not controller_class:
+            raise SqPollerConfError(f"Unknown plugin called {ptype}")
+
+        return [controller_class[ptype](plugin_conf)]
```
[the remark this answers is not in the document]
Comment (LucaNicosia) [reply suzieq/poller/controller/base_controller_plugin.py:34]: Each plugin perform its own validation and if the plugin_conf is wrong, the plugin itself will raise an exception.
Probably the only missing validation is `plugin_conf is not None`. We'll add this check

**Pull request merged**